lib/pmdtester/parsers/pmd_report_document.rb

```diff
@@ -36,22 +39,37 @@ def start_element(name, attrs = [])
 
     def characters(string)
       @current_violation.text = string unless @current_violation.nil?
-      @current_error.text = string unless @current_error.nil?
     end
 
     def end_element(name)
       case name
       when 'file'
-        @violations.add_violations_by_filename(@current_filename, @current_violations)
+        unless @current_violations.empty?
+          @violations.add_violations_by_filename(@current_filename, @current_violations)
+        end
         @current_filename = nil
       when 'violation'
-        @current_violations.push(@current_violation)
+        @current_violations.push(@current_violation) if match_filter_set?(@current_violation)
         @current_violation = nil
       when 'error'
         @errors.add_error_by_filename(@current_filename, @current_error)
         @current_filename = nil
         @current_error = nil
       end
     end
+
+    def cdata_block(string)
+      @current_error.text = string unless @current_error.nil?
+    end
+
+    def match_filter_set?(violation)
+      return true if @filter_set.nil?
+
+      @filter_set.each do |ruleset|
+        return true if ruleset.eql?(violation.attrs['ruleset'].delete(' ').downcase)
```

Comment:
That's a pity, here ends the convention... right, the ruleset names have unfortunately camel case with spaces between the words... ("Best Practices" <-> bestpractices). Yes, we'll just have to "normalize" it here...